### ISSUE TITLE
[JENKINS-47148] Add a11y support for textInput

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#4470]
<p>There is ATM no way to use TextInput and have support for a11y, since you cannot pass an id (which could be used in <label htmlFor="id"/>) nor can you pass aria-* properties.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tscherler.png" title="tscherler's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler">tscherler</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47148" target="_blank">Add a11y support for textInput</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nicu">nicu</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>component(s)</b>: blueocean-plugin
<li><b>label(s)</b>: testing-notes-reviewed
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2017-10-03T08:26:47+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>There is ATM no way to use TextInput and have support for a11y, since you cannot pass an id (which could be used in &lt;label htmlFor="id"/&gt;) nor can you pass aria-* properties.</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tscherler.png" title="tscherler's avatar" /> tscherler:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47148?focusedId=315744&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315744">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://github.com/jenkinsci/blueocean-plugin/pull/1428" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/1428</a></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://github.com/jenkinsci/blueocean-plugin/pull/1428" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/1428</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kshultz.png" title="kshultz's avatar" /> kshultz:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-47148?focusedId=317919&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-317919">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><b>Testing Notes:</b>
<a href="https://github.com/jenkinsci/blueocean-plugin/pull/1428/files#diff-2c253b38e784a0f2773343b91f594749" class="external-link" target="_blank" rel="nofollow noopener">A test case was added</a> as part of the above-mentioned PR.</p>
</pre>
</details>

</ul>
</details>

<p><b>Testing Notes:</b><br/>
<a href="https://github.com/jenkinsci/blueocean-plugin/pull/1428/files#diff-2c253b38e784a0f2773343b91f594749" class="external-link" target="_blank" rel="nofollow noopener">A test case was added</a> as part of the above-mentioned PR.</p>
